autonomous/autonomous.py

```diff
@@ -97,95 +81,80 @@ def setup(self) -> None:
         self.field_auto_target_pose.setPose(
             trajectory_generator.goal_to_field(Pose2d(0, 0, 0))
         )
+        self.trajectory_config = TrajectoryConfig(maxVelocity=3.5, maxAcceleration=9.0)
+        self.trajectory_config.addConstraint(
+            constraint.SwerveDrive4KinematicsConstraint(
+                self.chassis.kinematics, maxSpeed=4.0
```

Comment:
It limits the combined movement so that an individual module doesn't exceed its maximum speed.

Reply:
I'll note that the chassis also clamps module speed. It would be a good idea for these to agree with each other. I presume adding this constraint ensures that it doesn't attempt to generate a trajectory with impossible speeds, so the constraint should still be added.

https://github.com/thedropbears/pyrapidreact/blob/b6232ed209edd0566f35e129c95610101ff9eb4a/components/chassis.py#L141

https://github.com/thedropbears/pyrapidreact/blob/b6232ed209edd0566f35e129c95610101ff9eb4a/components/chassis.py#L250-L252

Reply:
Oliver's PR that added this constant is good - I'll use it here rather than a magic number.

Reply:
does the trajectory generator know about the actual rotation of the chassis though?

Reply:
That's true, it does not. Only the "rotation" caused by the curvature of the path.